llpc/context/llpcPipelineContext.cpp

```diff
@@ -764,11 +838,17 @@ void PipelineContext::setVertexInputDescriptions(Pipeline *pipeline) const {
 }
 
 // =====================================================================================================================
-// Set color export state in middle-end Pipeline object
+// Set color export state in middle-end Pipeline object, and/or hash it.
 //
-// @param pipeline : Pipeline object
-void PipelineContext::setColorExportState(Pipeline *pipeline) const {
+// @param [in/out] pipeline : Middle-end pipeline object; nullptr if only hashing
+// @param [in/out] hasher : Hasher object; nullptr if only setting LGC pipeline state
+void PipelineContext::setColorExportState(Pipeline *pipeline, Util::MetroHash64 *hasher) const {
```

Comment:
I don't like having a single entry point that does two different things.  It should be two functions.  Leave `setColorExportState` create a new function `hashColorExportState`.

Reply:
As above.